ouroboros-consensus-cardano/test/byron-test/Test/Consensus/Byron/Serialisation.hs

```diff
@@ -46,7 +47,7 @@ tests = testGroup "Byron"
         ]
     ]
   where
-    dictNestedHdr :: forall a. NestedCtxt_ ByronBlock Header a -> Dict (Eq a, Show a)
+    dictNestedHdr :: forall a. NestedCtxt_ ByronBlock Header a -> Dict (And Eq Show) a
```

Comment:
Minor: Thinking out loud: Do we actually want to use the `Data.SOP.Dict` everywhere? I think it is more natural to still use a "normal" `Dict` (eg from [`constraints`](https://hackage.haskell.org/package/constraints-0.14/docs/Data-Constraint.html#t:Dict)) in places like this, and only use the `sop-core` `Dict` when we have to partially apply it.

Also, this approach won't work *directly* (without sth like `Id` just below), eg for `Dict (F a)` where `F` is a type family.

Or instead of using a different `Dict`, just use `Dict'` defined like this in terms of the `sop-core` `Dict`:
```haskell
type Dict' c = Dict Id c

type Id :: Constraint -> Constraint
class    (c) => Id c
instance (c) => Id c
```

Reply:
Downgrading it to `constaints` Dicts is also fine for me. I just didn't want just another Dict type around

Reply:
Another idea: to avoid the perception of different kinds of `Dict`s, we could also enforce that eg the `sop-core` `Dict` always has to be imported qualified (via hlint?).

Reply:
>Also, this approach won't work directly (without sth like Id just below), eg for Dict (F a) where F is a type family.

I think we have a fairly heavy usage of type families in Consensus that would justify keeping the SOP Dict. Or maybe I'm misunderstanding you

Reply:
I mean that you can't rewrite `Data.Constraint.Dict (F a)` to `Data.SOP.Dict` without sth like `Id` AFAIK.

Reply:
Resolved to use `constraint` `Dict` by default, and only `sop-core` `Dict` when necessary and with a qualified import.

Reply:
Done